torch/csrc/jit/codegen/cuda/lower_sync_information.cpp

```diff
@@ -205,25 +211,25 @@ void SyncMap::build(Fusion* fusion) {
             continue;
           }
 
+          // In the case when the parallel id's are mapped by ca map,
+          //   will additionally need to consider if the producer is
+          //   a redundant write. The raw dim can be skipped only if
+          //   consumer use chains only contain redundant uses.
+          //  TODO:
+          //    still losing a bit precision here for expr ordering
+          //  sensitive cases, but we could wait until that becomes
+          //  a perf limiter to fix.
```

Comment:
This analysis is expression order insensitive so we'd need to incorporate this TODO into the sync insertion pass, which didn't seem to be trivial to do, essentially need to track the smem writes more carefully. 

As this case seems to be a rare case among already rare cases, in terms of perf benefit I think it could wait.